src/utilities/fastify/plugins/fastifyActiveEndpoint.spec.ts

```diff
@@ -28,17 +31,21 @@ describe('fastifyActiveEndpoint', () => {
 
   let activeEndpoint: Partial<InternetEndpoint>;
   beforeEach(() => {
+    mockEndpointManagerInit.mockResolvedValueOnce({
+      getActiveEndpoint: mockGetActiveEndpoint,
+    } as any)
     activeEndpoint = {
       makeInitialSessionKeyIfMissing: mockMakeInitialSessionKey,
     };
-    mockGetActiveEndpoint.mockResolvedValueOnce(activeEndpoint);
+    mockGetActiveEndpoint.mockResolvedValueOnce(activeEndpoint as any);
```

Comment:
Why mock this? Unless I made a mistake in the test util for the endpoint manager, this should just work without any mocks or spys.

Reply:
Tried this way but activeEndpoint is a different object then app.activeEndpoint since getActiveEndpoint always returns new instance
```
  const getConnection = setUpTestDbConnection();
  const getInternetEndpointManager = mockInternetEndpoint(getConnection);
  

  beforeEach(async () => {
    activeEndpoint = await getInternetEndpointManager().getActiveEndpoint();
  })

  test('Fastify should be decorated with active endpoint', async () => {
    const app = fastify();
    await app.register(fastifyMongoose);

    await app.register(fastifyActiveEndpoint);
    await app.ready();

    expect(app).toHaveProperty('activeEndpoint');
    expect(app.activeEndpoint === activeEndpoint).toBeTrue()
  });
  ```

Reply:
But why does it matter whether it's the same object? It should be more than enough to know it's equivalent:

```ts
expect(app.activeEndpoint).toMatchObject<InternetEndpoint>(jest.expectObjectContaining({
  id: endpoint.id,
  internetAddress: endpoint.internetAddress,
}));
```

When writing tests, we should be more interested in _outcomes_. Spies and mocks should be used sparingly -- when we can't (easily) inspect the outcomes.

Reply:
Hypothetically, in the future, we could decide to change the following:

```ts
const manager = await InternetEndpointManager.init(conn);
const endpoint = await manager.getActiveEndpoint();
```

to:

```ts
const endpoint = await InternetEndpoint.getActive(conn);
```

If we do that and we keep the tests you wrote originally, we'd have to re-implement a lot of testing code. But why? We're only interested in the outcome.

Reply:
One more thing: Even if we -- hypothetically -- wanted to verify that we're calling a method in a particular way, we wouldn't need to mock anything as done here.

We should do something like this:

```ts
// Plan A, if you have access to the instance
const spy = jest.spyOn(instance, 'method');

// Plan B, if you don't have access to the instance
const spy = jest.spyOn(Class.prototype, 'method');
```

... And you'd only do that in the test where you want to check how `Class.method()` is called.